generic_client.go

```diff
@@ -56,15 +59,19 @@ func (gc *GenericClient) DTORequest(pop DTO, method, path string, pathParams, qu
 }
 
 // Request performs an HTTP request and returns the body of the response
-func (gc *GenericClient) Request(method, path string, pathParams, queryParams urlParams, body ...DTO) (resBody io.ReadCloser, err error) {
+func (gc *GenericClient) Request(resourceName, method, path string, pathParams, queryParams urlParams, body ...DTO) (resBody io.ReadCloser, err error) {
 	req, err := gc.buildRequest(method, path, pathParams, queryParams, body...)
 	if err != nil {
 		return
 	}
+	start := time.Now()
 	res, err := gc.HTTP.Do(req)
 	if err != nil {
 		return
 	}
+
+	messages.ReportClientHTTPResponse(gc.Logger, res, resourceName, time.Now().Sub(start))
```

Comment:
See earlier comment; could this not be: `gc.LogFunc(res, resourceName, time.Now().Sub(start))`

To me this seems like a cheap change that avoids coupling but I could be missing something.